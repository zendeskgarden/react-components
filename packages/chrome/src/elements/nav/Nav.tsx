/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes } from 'react';
import PropTypes from 'prop-types';
import { useChromeContext } from '../../utils/useChromeContext';
import { NavContext } from '../../utils/useNavContext';
import { StyledNav } from '../../styled';

interface INavProps extends HTMLAttributes<HTMLElement> {
  /** Expands the nav area to display the item text */
  isExpanded?: boolean;
}

/**
 * @extends HTMLAttributes<HTMLElement>
 */
export const Nav = React.forwardRef<HTMLElement, INavProps>((props, ref) => {
  const { hue, isLight, isDark } = useChromeContext();
  const navContextValue = { isExpanded: !!props.isExpanded };

  return (
    <NavContext.Provider value={navContextValue}>
      <StyledNav ref={ref} {...props} hue={hue} isLight={isLight} isDark={isDark} />
    </NavContext.Provider>
  );
});

Nav.displayName = 'Nav';

Nav.propTypes = {
  isExpanded: PropTypes.bool
};
