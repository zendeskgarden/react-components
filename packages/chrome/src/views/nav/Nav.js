/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import PropTypes from 'prop-types';
import styled from 'styled-components';
import classNames from 'classnames';
import { retrieveTheme } from '@zendeskgarden/react-theming';
import ChromeStyles from '@zendeskgarden/css-chrome';

const COMPONENT_ID = 'chrome.nav';

/**
 * Accepts all `<div>` props
 */
const Nav = styled.nav.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': PACKAGE_VERSION,
  className: props =>
    classNames(ChromeStyles['c-chrome__nav'], {
      // State
      [ChromeStyles['c-chrome__nav--expanded']]: props.expanded,

      // Colors
      [ChromeStyles['c-chrome__nav--dark']]: props.dark,
      [ChromeStyles['c-chrome__nav--light']]: props.light
    })
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

Nav.propTypes = {
  /** Expand navigation area to include item text */
  expanded: PropTypes.bool,
  /** Apply dark styling */
  dark: PropTypes.bool,
  /** Apply light styling */
  light: PropTypes.bool
};

/** @component */
export default Nav;
