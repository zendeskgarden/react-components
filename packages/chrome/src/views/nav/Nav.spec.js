/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React from 'react';
import { shallow } from 'enzyme';
import Nav from './Nav';

describe('Nav', () => {
  it('renders default styling', () => {
    const wrapper = shallow(<Nav />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders expanded styling if provided', () => {
    const wrapper = shallow(<Nav expanded />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders dark styling if provided', () => {
    const wrapper = shallow(<Nav dark />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders light styling if provided', () => {
    const wrapper = shallow(<Nav light />);

    expect(wrapper).toMatchSnapshot();
  });
});
