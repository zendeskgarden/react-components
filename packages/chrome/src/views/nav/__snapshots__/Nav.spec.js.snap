// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders default styling 1`] = `
<nav
  className="c-chrome__nav "
  data-garden-id="chrome.nav"
  data-garden-version="version"
/>
`;
